pdns/dnspacket.cc

```diff
@@ -658,3 +673,27 @@ bool checkForCorrectTSIG(const DNSPacket* q, UeberBackend* B, string* keyname, s
 
   return result;
 }
+
+bool getTSIGHashEnum(const string &algoName, TSIGHashEnum& algoEnum)
```

Comment:
It would be cool if this piece of code (plus the enum) would be rewritten to an unordered map (as we're doing C++2011 anyway) , so you can use the `[]` operator to get the correct value from the map. Although that would require changes to e.g. the dnssecinfra.cc as well.